src/datasources/asset/AssetDataSource.ts

```diff
@@ -70,10 +96,203 @@ export class AssetDataSource extends DataSourceBase<AssetMetadataQuery> {
+    let entityIds: string[];
+    if (query.entityType === EntityType.Asset) {
+      entityIds = assets
+    } else {
+      entityIds = systems
+    }
+    // filter empty values
+    entityIds = entityIds.filter(Boolean);
+    // retrieve the first 10 asset or system IDs to display some data, ensuring the utilization graph is not empty.
+    if (!entityIds.length) {
+      if (query.entityType === EntityType.Asset) {
+        let conditions: string[] = [];
+        if (workspaceId) {
+          conditions.push(`workspace = "${workspaceId}"`);
+        }
+        if (systems.length) {
+          const systemsCondition = systems.map(id => `${AssetFilterProperties.LocationMinionId} = "${id}"`)
+          conditions.push(`(${systemsCondition.join(' or ')})`);
+        }
+        const assetFilter = conditions.join(' and ');
+        const assetsResponse = await this.queryAssets(assetFilter, 10);
+        entityIds = assetsResponse.map((asset) => {
+          return asset.id;
+        })
+      } else {
+        let filterString = '';
+        if (query.workspace) {
+          filterString += `workspace = "${query.workspace}"`;
+        }
+        let systemMetadata = await this.getSystems({
+          filter: filterString,
+          projection: `new(${defaultProjection.join()})`,
+          orderBy: defaultOrderBy,
+          take: 10
+        })
+        entityIds = systemMetadata.data.map((system) => {
+          return system.id;
+        })
+      }
+    }
+    const resultArray: Array<{
+      id: string,
+      datetimes: number[],
+      values: number[]
+    }> = [];
+    for (let index = 0; index < entityIds.length; index++) {
+      let data: AssetUtilizationHistory[] = [];
+      let requestCount = 0;
+      do {
+        try {
+          let requestBody: QueryAssetUtilizationHistoryRequest = {
+            utilizationFilter: '',
+            continuationToken: continuationToken,
+            orderBy: AssetUtilizationOrderBy.START_TIMESTAMP,
+            orderByDescending: true
+          }
+          if (query.entityType === EntityType.Asset) {
+            requestBody.assetFilter = `${AssetFilterProperties.AssetIdentifier} = "${entityIds[index]}"`;
+          } else {
+            requestBody.assetFilter = `${AssetFilterProperties.LocationMinionId} = "${entityIds[index]}" and ${AssetFilterProperties.IsSystemController} = true`;
+          }
+          const response: AssetUtilizationHistoryResponse = await this.post<AssetUtilizationHistoryResponse>(this.baseUrl + '/query-asset-utilization-history', requestBody);
+          requestCount++;
+          continuationToken = response.continuationToken || '';
+          data = data.concat(response.assetUtilizations);
+        } catch (error) {
+          throw new Error(`An error occurred while retrieving utilization history: ${error}`);
+        }
+      } while (
+        continuationToken && (requestCount < 25) &&
```

Comment:
I believe I had commented this on the last PR, but I'm concerned about the number of requests we're making here. If I'm reading it correctly, we allow up to 10 systems/assets and a make up to 25 requests for each one. This is an abnormally high number of requests to make from a UI. What determines how many requests we make here? Is it how large of a time range the user has selected?

Hoping @CiprianAnton or @kkerezsi can provide some suggestions on this.

Reply:
is this comment actual or I can close it? @mure @CiprianAnton @kkerezsi